mxcube3/routes/qutils.py

```diff
@@ -304,7 +304,15 @@ def _handle_dc(sample_id, node, include_lims_data=True):
         limsres = mxcube.rest_lims.get_dc(lims_id)
 
     # Always add link to data, (no request made)
-    limsres["limsTaskLink"] = mxcube.rest_lims.dc_link(lims_id)
```

Comment:
Id prefer if this was done slightly differently, either you implement your ISPyBRestClient HO, with the correct method, which perhaps is the preferred way (using exi or not your choice)

The other way would be to add a method in limsutils, for instance called get_dc_link that performs the right logic. There is also no need for the .use_exi attribute since you can simply check for mxcube.rest_lims if it does not exist
use db_connection. This would at lest the duplication of this logic ...

Id recommend you to use the ISPyBRestClient though, I think the long term strategy is to phase out the SOAP interface, so we are in a transition phase right now. Another question is why you would not use EXI ?

Marcus

Reply:
We are using both soap and rest hwobj. But, the dc_link in the rest hwobj only works with exi like urls. Thats why I cannot check in in the rest hwobj.

I can add a method in limsutils but I still need a selector somewhere.

We do not have exi in place here, it is not straightforward deployment although we have a test run. We have seen things here and there and we have enough with one ispyb flavour for now...

Reply:
Ah, so maybe we can add some configuration attributes to the ISPyBRestClient, now there is EXI root:

  <exi_root>
   https://your.exi.org
  </exi_root>

  <restRoot>
   https://your.lims.org
  </restRoot>

in lims-rest.xml

Maybe this can be changed to something more generic for the dc_link method or you can just use that to point at another url for the link ?

like:

  <result_link_root>
   ...
  </result_link_root>

It would be more generic I think

Reply:
what I do not like on that approach is that in our case we do not use the rest interface for getting the link, so I feel like it should be on the soap hwobj, which is anyway something missing now.

Reply:
Ok, so then something like this

- Add 

<result_link_root>
...
</result_link_root>

To lims.xml (the SOAP HO) and maybe also to lims-rest.xml (Rest HO)

- Add a method in limsutils called get_result_link(lims_id),
That first tries to get the one from the Rest HO and if nothing is returned the one from the SOAP 

HO. Something like that.

Reply:
The dc_link method in the rest_hwobj always returns a url, not None. Which points to a non existing page in our case. Thats I only see a selector flag as a solution (well, move everything to rest, but thats the future ;))